FIPS: Failures in SAML tests

When running testsuite with Keycloak on Quarkus distribution with FIPS support and BCFIPS security providers (after #14886), there are failures like those below. There are really lots of failures and hence it should be investigated whether to divide this issue into more issues.

Some snippets from test failures:
```
[ERROR]   SAMLServletAdapterTest.salesPostEncRejectConsent:876 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, X-Content-Type-Options: nosniff, X-XSS-Protection: 1; mode=block, Content-Type: application/json, content-length: 25] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 25,Chunked: false]}}> with entity {"error":"unknown_error"}
[ERROR]   SAMLServletAdapterTest.salesPostSigChangeContents:1033 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, X-Content-Type-Options: nosniff, X-XSS-Protection: 1; mode=block, Content-Type: application/json, content-length: 25] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 25,Chunked: false]}}> with entity {"error":"unknown_error"}
[ERROR]   SAMLServletAdapterTest.salesPostSigStaxParsingFlawEmailTest:1012 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, X-Content-Type-Options: nosniff, X-XSS-Protection: 1; mode=block, Content-Type: application/json, content-length: 25] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 25,Chunked: false]}}> with entity {"error":"unknown_error"}

[ERROR]   SamlSignatureTest.testNoChange:330->testSamlResponseModifications:272->testSamlResponseModificationsClient:292 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, Cache-Control: no-store, must-revalidate, max-age=0, X-Content-Type-Options: nosniff, Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/demo/; Secure; HttpOnly, Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/demo/; Secure; HttpOnly, Set-Cookie:

[ERROR]   SamlSignatureTest.testXSW5:360->testSamlResponseModifications:272->testSamlResponseModificationsClient:292 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, Cache-Control: no-store, must-revalidate, max-age=0, X-Content-Type-Options: nosniff, Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/demo/; Secure; HttpOnly, Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/demo/; Secure; HttpOnly, Set-Cookie: KEYCLOAK_IDENTITY=eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJjNWJlOTU0MS0yZDcyLTQ4ZDgtYTZjOC1lMDY0OGYzZTFkNjcifQ.eyJleHAiOjE2NjU1NTAyNzUsImlhdCI6MTY2NTUxNDI3NSwianRpIjoiMDIwYTBjN2ItMTlmNy00MzEwLWJiNjctNjM3YjViZWViZTQ1IiwiaXNzIjoiaHR0cHM6Ly9sb2NhbGhvc3Q6ODU0My9hdXRoL3JlYWxtcy9kZW1vIiwic3ViIjoiYWJmMjViZGEtMDgzMC00MzQzLWE5MzYtNThlNWI1NDAyOWZjIiwidHlwIjoiU2VyaWFsaXplZC1JRCIsInNlc3Npb25fc3RhdGUiOiJlYjc1YTE4YS02NzdiLTQ5MzUtOTczNy0wMjNmNmI2YzRhMzYiLCJzaWQiOiJlYjc1YTE4YS02NzdiLTQ5MzUtOTczNy0wMjNmNmI2YzRhMzYiLCJzdGF0ZV9jaGVja2VyIjoiUmxjMENDRWxhZ05HRnFhTkpjTnY2dTVJRjd0TGljSU1vbWNDQU11Y3A4ayJ9.Kh1Z5BjIri5BWskcNSHsQZ9z0lzwZNVC2Np3ep1kNf4; Version=1; Path=/auth/realms/demo/; SameSite=None; Secure; HttpOnly, Set-Cookie: KEYCLOAK_IDENTITY_LEGACY=eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJjNWJlOTU0MS0yZDcyLTQ4ZDgtYTZjOC1lMDY0OGYzZTFkNjcifQ.eyJleHAiOjE2NjU1NTAyNzUsImlhdCI6MTY2NTUxNDI3NSwianRpIjoiMDIwYTBjN2ItMTlmNy00MzEwLWJiNjctNjM3YjViZWViZTQ1IiwiaXNzIjoiaHR0cHM6Ly9sb2NhbGhvc3Q6ODU0My9hdXRoL3JlYWxtcy9kZW1vIiwic3ViIjoiYWJmMjViZGEtMDgzMC00MzQzLWE5MzYtNThlNWI1NDAyOWZjIiwidHlwIjoiU2VyaWFsaXplZC1JRCIsInNlc3Npb25fc3RhdGUiOiJlYjc1YTE4YS02NzdiLTQ5MzUtOTczNy0wMjNmNmI2YzRhMzYiLCJzaWQiOiJlYjc1YTE4YS02NzdiLTQ5MzUtOTczNy0wMjNmNmI2YzRhMzYiLCJzdGF0ZV9jaGVja2VyIjoiUmxjMENDRWxhZ05HRnFhTkpjTnY2dTVJRjd0TGljSU1vbWNDQU11Y3A4ayJ9.Kh1Z5BjIri5BWskcNSHsQZ9z0lzwZNVC2Np3ep1kNf4; Version=1; Path=/auth/realms/demo/; Secure; HttpOnly, Set-Cookie: KEYCLOAK_SESSION=demo/abf25bda-0830-4343-a936-58e5b54029fc/eb75a18a-677b-4935-9737-023f6b6c4a36; Version=1; Expires=Wed, 12-Oct-2022 04:51:15 GMT; Max-Age=36000; Path=/auth/realms/demo/; SameSite=None; Secure, Set-Cookie: KEYCLOAK_SESSION_LEGACY=demo/abf25bda-0830-4343-a936-58e5b54029fc/eb75a18a-677b-4935-9737-023f6b6c4a36; Version=1; Expires=Wed, 12-Oct-2022 04:51:15 GMT; Max-Age=36000; Path=/auth/realms/demo/; Secure, Set-Cookie: KEYCLOAK_REMEMBER_ME=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/demo/; Secure; HttpOnly, P3P: CP="This is not a P3P policy!", X-XSS-Protection: 1; mode=block, Content-Type: application/json, content-length: 25] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 25,Chunked: false]}}> with entity {"error":"unknown_error"}
[ERROR]   SamlSignatureTest.testXSW6:365->testSamlResponseModifications:272->testSamlResponseModificationsClient:292 
Expected: response status code matches is <200>

[ERROR]   IdentityProviderTest.testSamlExportSignatureOn:1069 expected:<200> but was:<500>

[ERROR]   KcSamlBrokerAllowedClockSkewTest.loginClientExpiredResponseFromIdPWithClockSkew:101 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, X-Content-Type-Options: nosniff, Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/consumer/; HttpOnly, Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/consumer/; HttpOnly, Set-Cookie: ...

[ERROR]   KcSamlSignedBrokerTest.testSignatureTampering_signDoc_signAssert_encAssert:352->loginAttackChangeSignature:387->loginAttackChangeSignature:432->withSignedEncryptedAssertions:90->lambda$loginAttackChangeSignature$3:447 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Referrer-Policy: no-referrer, X-Frame-Options: SAMEORIGIN, Strict-Transport-Security: max-age=31536000; includeSubDomains, Cache-Control: no-store, must-revalidate, max-age=0, X-Content-Type-Options: nosniff, Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/provider/; HttpOnly, Set-Cookie: KC_RESTART=; Version=1; ...

[ERROR]   RealmTest.convertSAMLClientDescription:621 » InternalServerError HTTP 500 Internal Server Error
[ERROR]   KcSamlBrokerFrontendUrlTest.testFrontendUrlInDestinationExpected » Runtime java.lang.AssertionError: 
Expected: response status code matches is <200>
     but: was <HttpResponseProxy{HTTP/1.1 500 Internal Server Error [Set-Cookie: KEYCLOAK_LOCALE=; Version=1; Comment=Expiring cookie; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/realms/consumer/; HttpOnly, Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/realms/consumer/; HttpOnly, Set-Cookie: ...
...

```

--- a/testsuite/integration-arquillian/servers/auth-server/common/fips/kc.java.security
+++ b/testsuite/integration-arquillian/servers/auth-server/common/fips/kc.java.security
@@ -15,13 +15,17 @@
 #
 # Security providers used when global crypto-policies are set to FIPS (Usually it is used when FIPS enabled on system/JVM level)
 #
-#fips.provider.1=SunPKCS11 ${java.home}/conf/security/nss.fips.cfg
-#fips.provider.2=SUN
-#fips.provider.3=SunEC
-#fips.provider.4=com.sun.net.ssl.internal.ssl.Provider SunPKCS11-NSS-FIPS
-#fips.provider.5=SunJGSS
-#fips.provider.6=XMLDSig
-#fips.provider.5=
+# NOTE: This list of providers is needed to be override just because XMLDSig provider is not yet present on the OpenJDK 17 by default on the RHEL FIPS host on OpenJDK 17.0.3.
+# However once it is present, there won't be a need to override this and this part can be fully commented/removed.
+# TODO: Comment/remove this once https://bugzilla.redhat.com/show_bug.cgi?id=1940064 is fixed and OpenJDK 17 updated to corresponding version where XMLDSig is available by default
+#
+fips.provider.1=SunPKCS11 ${java.home}/conf/security/nss.fips.cfg
+fips.provider.2=SUN
+fips.provider.3=SunEC
+fips.provider.4=SunJSSE
+fips.provider.5=SunJCE
+fips.provider.6=SunRsaSign
+fips.provider.7=XMLDSig
 
 # Commented this provider for now (and also other providers) as it uses lots of non-FIPS services.
 # See https://access.redhat.com/documentation/en-us/openjdk/11/html-single/configuring_openjdk_11_on_rhel_with_fips/index#ref_openjdk-default-fips-configuration_openjdk

--- a/testsuite/integration-arquillian/tests/base/testsuites/fips-suite
+++ b/testsuite/integration-arquillian/tests/base/testsuites/fips-suite
@@ -13,3 +13,8 @@ KcRegTest
 KcRegCreateTest
 KcAdmTest
 KcAdmCreateTest
+SAMLServletAdapterTest
+SamlSignatureTest
+KcSamlBrokerTest
+KcSamlFirstBrokerLoginTest
+KcSamlEncryptedIdTest

